boot-speed/clouds/measure-cloud.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+"""
+Measure the boot speed of cloud instances.
+
+Copyright 2019 Canonical Ltd.
+Paride Legovini <paride.legovini@canonical.com>
+"""
+
+import argparse
+import datetime as dt
+import glob
+import json
+import logging
+import tarfile
+import os
+import paramiko
+import pycloudlib
+import sys
+
+from pathlib import Path
+
+
+known_clouds = ['ec2', 'gce']
+
+
+def create_sftp_client(host, user, port=22, password=None, keyfilepath=None):
+    """
+    create_sftp_client(host, port, user, password, keyfilepath) -> SFTPClient
+    """
+    sftp = None
+    key = None
+    transport = None
+    try:
+        if keyfilepath is not None:
+            # Get private key used to authenticate user.
+            key = paramiko.RSAKey.from_private_key_file(keyfilepath)
+
+        # Create Transport object using supplied method of authentication.
+        transport = paramiko.Transport((host, port))
+        transport.connect(None, user, password, key)
+
+        sftp = paramiko.SFTPClient.from_transport(transport)
+
+        return sftp
+    except Exception as e:
+        print('An creating SFTP client: %s: %s' % (e.__class__, e))
+        if sftp is not None:
+            sftp.close()
+        if transport is not None:
+            transport.close()
+        pass
+
+
+def cloudid2serial(id):
```

Comment:
we should consider adding this to pycloudlib

Reply:
+1, it belongs there, but I'd like to see the job working first, to be sure there are no bigger issues around. I'll have to start working on pycloudlib to add the missing clouds at some point, that will probably be a good moment to add this function too.